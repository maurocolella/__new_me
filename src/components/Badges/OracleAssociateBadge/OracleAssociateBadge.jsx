import React, { PureComponent } from 'react';
import ReactGA from 'react-ga';
import OracleAssociateImg from '../../../assets/images/oracle-certified-associate-java-se-8-programmer.png';

import styles from '../styles.scss';

class OracleAssociateBadge extends PureComponent {
  popup = (event) => {
    event.preventDefault();
    event.stopPropagation();
    const url = event.currentTarget.getAttribute('href');
    const urlHash = btoa(url);

    const cookiesAccepted = localStorage ? localStorage.getItem('cookiesAccepted') : false;
    if (cookiesAccepted) {
      ReactGA.outboundLink(
        {
          label: url,
        },
        () => {},
      );
    }

    window.open(url, urlHash);
  }

  render() {
    return (
      <a
        href="https://www.youracclaim.com/badges/82b8a687-2958-4e07-b34d-9fae29eb9a62/public_url"
        onClick={this.popup}
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src={OracleAssociateImg}
          alt="Oracle Certified Associate, Java SE 8 Programmer"
          className={styles.badge}
        />
      </a>
    );
  }
}

export default OracleAssociateBadge;
