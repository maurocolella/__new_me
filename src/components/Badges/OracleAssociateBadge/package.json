{
  "name": "OracleAssociateBadge",
  "version": "0.0.0",
  "private": true,
  "main": "./OracleAssociateBadge.jsx"
}
