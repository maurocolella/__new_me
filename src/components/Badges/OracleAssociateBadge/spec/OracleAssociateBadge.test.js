import React from 'react';
import renderer from 'react-test-renderer';
import OracleAssociateBadge from '../OracleAssociateBadge';

it('renders correctly', () => {
  const tree = renderer
    .create(<OracleAssociateBadge />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
