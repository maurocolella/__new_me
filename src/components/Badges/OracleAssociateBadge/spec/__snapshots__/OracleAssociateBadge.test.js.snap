// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-iframe-height="270"
  data-iframe-width="150"
  data-share-badge-id="82b8a687-2958-4e07-b34d-9fae29eb9a62"
/>
`;
